applications/AdjointFluidApplication/custom_processes/output_primal_solution_process.h

```diff
@@ -316,6 +331,158 @@ class OutputPrimalSolutionProcess : public Process
 
     void ExecuteFinalize() override
     {
+        // write the values to the file after finishing the computation
+        KRATOS_TRY
+
+        if( mNodalValues.size() > 0 || mProcessInfoValues.size() > 0 )
+        {
+            // open file to append data
+            H5::H5File file(mFilename.c_str(), H5F_ACC_RDWR);
+            
+            H5::DataSet NodeIdDataset = file.openDataSet("/NodalData/Id");
+
+            hsize_t dims[3];
+            NodeIdDataset.getSpace().getSimpleExtentDims(dims);
+            
+            if (mNumNodes != dims[0])
+                KRATOS_THROW_ERROR(std::runtime_error, "inconsistent dimension in file: ", mFilename)
+
+            if (mNumNodes != static_cast<SizeType>(mrModelPart.Nodes().size()))
+                KRATOS_THROW_ERROR(std::runtime_error, "detected change in number of nodes.", "")
+
+            std::vector<unsigned int> NodeIdBuffer(mrModelPart.Nodes().size());
+            NodeIdDataset.read(NodeIdBuffer.data(), H5::PredType::NATIVE_UINT);
+
+            // check that the node ordering is consistent between the model part and file
+            int Delta = 0;
+            auto itNode = std::begin(NodeIdBuffer);
+            std::for_each(std::begin(mrModelPart.Nodes()), std::end(mrModelPart.Nodes()),
+                [&](NodeType& rNode)
+                {
+                    Delta += std::abs(static_cast<int>(rNode.Id()) - static_cast<int>(*itNode));
+                    itNode++;
+                });
+
+            if (Delta != 0)
+                KRATOS_THROW_ERROR(std::runtime_error, "detected mismatch of node ids in file: ", mFilename);
```

Comment:
here you could use KRATOS_ERROR_IF(Delta != 0) << "detected ... " << mFilename << std::endl;